source/speech.py

```diff
@@ -313,11 +313,9 @@ def speakObjectProperties(obj,reason=controlTypes.REASON_QUERY,index=None,**allo
 def speakObject(obj,reason=controlTypes.REASON_QUERY,index=None):
 	from NVDAObjects import NVDAObjectTextInfo
 	role=obj.role
-	isEditable=(reason!=controlTypes.REASON_FOCUSENTERED and obj.TextInfo!=NVDAObjectTextInfo and (role in (controlTypes.ROLE_EDITABLETEXT,controlTypes.ROLE_TERMINAL,controlTypes.ROLE_DOCUMENT) or controlTypes.STATE_EDITABLE in obj.states))
-	# Browse mode document treeInterceptors currently handle announcing text content themselves, thus we should not class these as editable for deciding whether to announce text content here.
+	# If an NVDAObject has text that is navigable, and browse mode is not in use, then present the current line in speech
 	import browseMode
-	if isEditable and isinstance(obj.treeInterceptor,browseMode.BrowseModeDocumentTreeInterceptor) and obj is obj.treeInterceptor.rootNVDAObject:
-		isEditable=False
+	isEditable=(not isinstance(obj.treeInterceptor,browseMode.BrowseModeDocumentTreeInterceptor) or obj.treeInterceptor.passThrough) and obj._hasNavigableText

```

Comment:
I would be tempted to extract the first part of the `and` to a `isBrowseMode()` method